### PR TITLE
refactor slice_to_num_chips to adapt to Cloud config

### PR DESCRIPTION
**Pull request by zhihaoshan-google**

### CODE REVIEW

File: jetstream/core/config_lib.py [hunk 1]
```diff
@@ -85,11 +84,16 @@ class InterleavedCPUTestServer(ServerConfig):
 
 
 def slice_to_num_chips(s: str) -> int:
-  """Converts a TPU spec like v5e=4x2 to the number of chips, 8."""
-  # Account for the case where it is written 'v5e:4x2'.
-  delim = "=" if "=" in s else ":"
-  i = math.prod([int(c) for c in s.split(delim)[1].split("x")])
-  return i
+  """Converts a TPU spec like v5e-8 or v5e=8 to the number of chips, 8."""
```
Comment (FanhaiLu1) [inline jetstream/core/config_lib.py:87]: The new logic is more clear!

File: jetstream/core/config_lib.py [hunk 2]
```diff
@@ -85,11 +84,16 @@ class InterleavedCPUTestServer(ServerConfig):
...
+  # Account for the case where it is written 'tpu=8' for compatibility.
+  delim = "-" if "-" in s else "="
+  # TODO: Support more accelerator type check.
+  accelerator_type, num_devices = s.split(delim)
+  return (
+      int(num_devices)
+      if accelerator_type != "v4"
+      else int(int(num_devices) / 2)
```
Comment (FanhaiLu1) [inline jetstream/core/config_lib.py:95]: Nit: please use int(num_devices) // 2, you can save the one int convert

File: jetstream/core/config_lib.py [hunk 3]
```diff
@@ -85,11 +84,16 @@ class InterleavedCPUTestServer(ServerConfig):
...
-  i = math.prod([int(c) for c in s.split(delim)[1].split("x")])
-  return i
+  """Converts a TPU spec like v5e-8 or v5e=8 to the number of chips, 8."""
+  # Account for the case where it is written 'tpu=8' for compatibility.
+  delim = "-" if "-" in s else "="
+  # TODO: Support more accelerator type check.
+  accelerator_type, num_devices = s.split(delim)
+  return (
+      int(num_devices)
+      if accelerator_type != "v4"
```
Comment (FanhaiLu1) [inline jetstream/core/config_lib.py:94]: Is there any reason to only return half of chips?

**Review by FanhaiLu1 — APPROVED** [review]
Great, the new logic is more clear.

File: jetstream/core/config_lib.py (hunk 3 above)
Comment (zhihaoshan-google) [reply jetstream/core/config_lib.py]: V4-8 has 8 tensor cores but 4 chips (2 cores per chip)

File: jetstream/core/config_lib.py (hunk 2 above)
Comment (zhihaoshan-google) [reply jetstream/core/config_lib.py]: Thanks!

**Pull request merged**